source/common/ssl/context_impl.cc

```diff
@@ -469,12 +480,79 @@ ServerContextImpl::ServerContextImpl(ContextManagerImpl& parent, Stats::Scope& s
     RELEASE_ASSERT(rc == 1);
   }
 
+  // Hash configured SNIs for this context, so that sessions cannot be resumed across different
+  // filter chains, even when using the same server certificate.
+  for (const auto& name : server_names_) {
+    rc = EVP_DigestUpdate(&md, name.data(), name.size());
+    RELEASE_ASSERT(rc == 1);
+  }
+
   rc = EVP_DigestFinal(&md, session_context_buf, &session_context_len);
   RELEASE_ASSERT(rc == 1);
   rc = SSL_CTX_set_session_id_context(ctx_.get(), session_context_buf, session_context_len);
   RELEASE_ASSERT(rc == 1);
 }
 
+ssl_select_cert_result_t
+ServerContextImpl::processClientHello(const SSL_CLIENT_HELLO* client_hello) {
+  std::string server_name = EMPTY_STRING;
```

Comment:
nit: std::string default to empty, don't need init.

Reply:
Done.